src/pages/Forever/components/RedeemCode.tsx

```diff
@@ -1,100 +1,131 @@
 import styled from '@emotion/styled'
 import { colorsV3 } from '@hedviginsurance/brand'
 import { Button } from 'components/buttons'
-import { HedvigLogo } from 'components/icons/HedvigLogo'
 import { InputField } from 'components/inputs'
 import { Form, Formik } from 'formik'
 import React from 'react'
 import { useTextKeys } from 'utils/hooks/useTextKeys'
+import * as Yup from 'yup'
 
 interface RedeemCodeProps {
   code?: string
-  currentLocale: string
 }
 
-const PageWrapper = styled.div`
+const codeSchema = Yup.object({
+  code: Yup.string().required('FOREVER_CODE_ERROR'),
+})
+
+const INPUT_MAX_WIDTH = '21rem'
+
+const Wrapper = styled.div`
   display: flex;
   flex-direction: column;
-  justify-content: space-between;
-  height: 100vh;
-  color: ${colorsV3.gray500};
-  background-color: ${colorsV3.gray900};
+  flex-grow: 1;
 `
 
-const Header = styled.header`
+const RedeemForm = styled(Form)`
   display: flex;
-  justify-content: center;
-  padding-top: 1rem;
-  padding-bottom: 1rem;
+  flex-direction: column;
+  flex-grow: 1;
+
+  @media (min-width: 800px) {
+    justify-content: center;
+  }
 `
 
 const Main = styled.div`
-  padding: 0 1.625rem;
+  width: 100%;
+  margin-top: auto;
+  margin-bottom: auto;
+
+  @media (min-width: 800px) {
+    margin: 0;
+  }
 `
 
 const CodeField = styled.div`
   margin: 0 auto;
 
   @media (min-width: 480px) {
-    max-width: 325px;
+    max-width: ${INPUT_MAX_WIDTH};
+    margin-bottom: 1.125rem;
   }
 
   input {
+    width: 100%;
     padding-right: 0;
     text-align: center;
+    text-transform: uppercase;
   }
 `
 
 const Footer = styled.div`
   display: flex;
+  flex-direction: column;
   justify-content: center;
-  padding: 1rem 1.625rem 2.5rem;
-`
-
-const LogoLink = styled.a`
-  color: ${colorsV3.gray100};
+  align-items: center;
+  padding: 1rem 0 2.5rem;
 `
 
 const Paragraph = styled.p`
   text-align: center;
   color: ${colorsV3.gray500};
   font-size: 1rem;
-  margin-left: 1rem;
-  margin-right: 1rem;
+
+  @media (min-width: 800px) {
+    margin-bottom: 1.25rem;
+  }
 `
-const SubmitButton = styled(Button)`
+
+const SubmitButton = styled(Button)<{ disabled?: boolean }>`
   width: 100%;
-  min-width: 12rem;
   height: 3.5rem;
+  margin: 0 auto 1rem;
+  font-size: 1rem;
+  color: ${(props) => (props.disabled ? colorsV3.gray500 : colorsV3.gray900)};
+  background-color: ${(props) =>
+    props.disabled ? colorsV3.gray800 : colorsV3.purple500};
 
   @media (min-width: 480px) {
-    width: auto;
+    max-width: ${INPUT_MAX_WIDTH};
+  }
+`
+
+const Info = styled.p`
+  max-width: 25rem;
```

Comment:
`calc(${INPUT_MAX_WIDTH} + 2 * 2rem)` or something maybe makes it clearer how this is connected?